spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/domain/entities/Application.java

```diff
@@ -44,8 +50,28 @@
 
 	@lombok.Builder(builderClassName = "Builder", toBuilder = true)
 	private Application(String name, @Nullable BuildVersion buildVersion, @Nullable String status,
-			@Nullable Instant statusTimestamp, List<Instance> instances) {
+			@Nullable Instant statusTimestamp, List<Instance> instances, String groupName) {
+		Assert.notNull(name, "'name' must not be null");
+		Assert.hasText(groupName, "groupName must not be null");
```

Comment:
`name` is checked using `notNull()` but `groupName` is checked using `hasText()`. What is the difference?

Reply:
After I thought over it, the groupName can be nullable, because the legacy code doesn't have a groupName. I want to adopt new code to legacy code.